### PR TITLE
Added ECS managed scale in for ECS Cluster.

### PR DESCRIPTION
**Pull request by anuranduttaroy**
**Issue #:** NA

**Description of changes:**
Enable managed scaling & managed termination protection. Read more: https://docs.aws.amazon.com/AmazonECS/latest/developerguide/cluster-auto-scaling.html#asg-capacity-providers

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: deploy.js [hunk 1]
```diff
@@ -129,10 +130,59 @@ spawnOrFail('sam', ['package', '--s3-bucket', `${bucket}`,
...
+spawnOrFail('aws', ['autoscaling', 'set-instance-protection', 
+                    '--auto-scaling-group-name', `${autoScalingGroupName}`, 
+                    '--protected-from-scale-in', '--instance-ids', `${instanceIds[0]}`, `${instanceIds[1]}`]);
+
+// Create a capacity provider with managed scale-in
+let capacityProviderParam = {
+  autoScalingGroupArn: autoScalingGroupArn,
+  managedScaling: {
+    status: "ENABLED",
+    targetCapacity: 60,
```
Comment (ziyiz-amzn) [inline deploy.js:172]: Curious how does 60 comes, is it default value?

File: deploy.js [hunk 2]
```diff
@@ -129,10 +130,59 @@ spawnOrFail('sam', ['package', '--s3-bucket', `${bucket}`,
...
+const autoScalingGroupArn = asg.AutoScalingGroups && asg.AutoScalingGroups[0].AutoScalingGroupARN;
+const autoScalingGroupCapacityProviderName = autoScalingGroupName + 'CapacityProvider';
+var instanceIds = [];
+autoScalingGroupInstances.forEach(instance => {
+  instanceIds.push(instance.InstanceId);
+});
+
+spawnOrFail('aws', ['autoscaling', 'set-instance-protection', 
+                    '--auto-scaling-group-name', `${autoScalingGroupName}`, 
+                    '--protected-from-scale-in', '--instance-ids', `${instanceIds[0]}`, `${instanceIds[1]}`]);
```
Comment (ziyiz-amzn) [inline deploy.js:165]: May think of a dynamic way to list instance ids if asg desired capacity changes from 2 to other value.

File: deploy.js (hunk 1 above)
Comment (anuranduttaroy) [reply deploy.js:172]: Synced with you offline.

File: deploy.js (hunk 2 above)
Comment (anuranduttaroy) [reply deploy.js:165]: Synced with you offline.

**Pull request merged**